lib/imapSimple.js

```diff
@@ -175,6 +176,40 @@ ImapSimple.prototype.search = function (searchCriteria, fetchOptions, callback)
 };
 
 /**
+ * Download a "part" (either the message body, or an attachment)
+ */
+ImapSimple.prototype.getPartData = function(message, part) {
```

Comment:
this method is great! Retrieving the structure of the body, then downloading only the parts of the body that you want is a nice way of retrieving the message.